website/client/components/settings/verifyUsername.vue

```diff
@@ -0,0 +1,270 @@
+              v-model='temporaryUsername',
+              @blur='restoreEmptyUsername()',
+              :class='{"is-invalid input-invalid": usernameInvalid, "input-valid": usernameValid, "text-darker": temporaryUsername.length > 0}')
+    .mb-3(v-if="usernameIssues.length > 0")
+      .input-error.text-center(v-for="issue in usernameIssues") {{ issue }}
+    .small.text-center {{ $t('usernameLimitations') }}
+    .row.justify-content-center
+      button.btn.btn-primary(type='submit', @click='submitNames()' :disabled='usernameCannotSubmit') {{ $t('saveAndConfirm') }}
+    .scene_veteran_pets.center-block
+    .small.text-center.mb-3 {{ $t('verifyUsernameVeteranPet') }}
+    .small.text-center.tos-footer(v-html="$t('usernameTOSRequirements')")
+</template>
+
+<style lang="scss">
+  #verify-username___BV_modal_outer_ {
+    .modal-content {
+      height: 100%;
+      width: 566px;
+      padding-left: 2rem;
+      padding-right: 2rem;
+    }
+    .modal-dialog   {
+      -webkit-transform: translate(-5%, calc(50vh - 60%));
+      -ms-transform: translate(0, 50vh) translate(-5%, -60%);
+      -o-transform: translate(-5%, calc(50vh - 60%));
+      transform: translate(0, 50vh) translate(-5%, -60%);
+    }
+  }
+</style>
+
+<style lang="scss" scoped>
+  @import '~client/assets/scss/colors.scss';
+
+  button {
+    margin: 1rem;
+  }
+
+  .center-block {
+    margin: 0 auto 1em auto;
+  }
+
+  .col-3 {
+    padding-right: 0rem;
+  }
+
+  .form-group {
+    background-color: $gray-700;
+    border-radius: 2px;
+    border: solid 1px $gray-500;
+  }
+
+  h2 {
+    color: $purple-200;
+    margin-top: 1.5rem;
+  }
+
+  input {
+    border: 0px;
+  }
+
+  .input-error {
+    color: $red-50;
+    font-size: 90%;
+    width: 100%;
+  }
+
+  .input-group-prepend {
+    margin-right: 0px;
+  }
+
+  .input-group-text {
+    border: 0px;
+    background-color: $white;
+    color: $gray-300;
+    padding: 0rem 0rem 0rem 0.75rem;
+  }
+
+  label {
+    color: $gray-100;
+    font-weight: bold;
+    margin-bottom: 0rem;
+    margin-left: 1rem;
+  }
+
+  .nametag-header {
+    background-color: $gray-700;
+    border-radius: 0.3rem 0.3rem 0rem 0rem;
+    margin-left: -3rem;
+    margin-right: -3rem;
+    padding: 1rem 9rem 1rem 9rem;
+  }
+
+  p {
+    color: #686274;
+  }
+
+  .small {
+    color: $gray-200;
+  }
+
+  .text-darker {
+    color: $gray-50;
+  }
+
+  .tos-footer {
+    background-color: $gray-700;
+    border-radius: 0rem 0rem 0.3rem 0.3rem;
+    margin-left: -3rem;
+    margin-right: -3rem;
+    padding: 1rem 4rem 1rem 4rem;
+  }
+
+  #username {
+    padding-left: 0.25rem;
+  }
+
+</style>
+
+<script>
+  import axios from 'axios';
+  import debounce from 'lodash/debounce';
+  import helloNametag from 'assets/svg/hello-habitican.svg';
+  import { mapState } from 'client/libs/store';
+
+  export default {
+    computed: {
+      ...mapState({
+        user: 'user.data',
+      }),
+      displayNameInvalid () {
```

Comment:
I think usernames and display names with a lenght of 1 should be valid, here they're valid only with at least 2 chars